From 5ca8a502a67229af5580f2885bc14d4ee03fbc1a Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Tue, 3 Sep 2024 17:02:06 -0400
Subject: [PATCH 1/2] thunderbird: set MOZ_* variables for legacy profiles.ini

This fixes app startup for installations that are not using
`wrapThunderbird` (e.g. app from `homebrew`).

These variables are already set for nixpkgs thunderbird using
wrapThunderbird wrapper.

This is most helpful for Darwin users. But it will help with an
(unusual) scenario where Home Manager is used in a non-NixOS environment
with the app managed by another package manager, or installed directly
from Mozilla builds.
---
 modules/programs/thunderbird.nix | 26 +++++++++-----------------
 1 file changed, 9 insertions(+), 17 deletions(-)

diff --git a/modules/programs/thunderbird.nix b/modules/programs/thunderbird.nix
index c15a946ec663..b9a585d9cc7d 100644
--- a/modules/programs/thunderbird.nix
+++ b/modules/programs/thunderbird.nix
@@ -234,11 +234,11 @@ in {
         type = types.bool;
         default = true;
         example = false;
-        visible = isDarwin;
+        visible = false;
         readOnly = !isDarwin;
         description = ''
-          Warn to set environment variables before using this module. Only
-          relevant on Darwin.
+          Using programs.thunderbird.darwinSetupWarning is deprecated. The
+          module is compatible with all Thunderbird installations.
         '';
       };
     };
@@ -324,20 +324,6 @@ in {
       })
     ];
 
-    warnings = optional (isDarwin && cfg.darwinSetupWarning) ''
-      Thunderbird packages are not yet supported on Darwin. You can still use
-      this module to manage your accounts and profiles by setting
-      'programs.thunderbird.package' to a dummy value, for example using
-      'pkgs.runCommand'.
-
-      Note that this module requires you to set the following environment
-      variables when using an installation of Thunderbird that is not provided
-      by Nix:
-
-          export MOZ_LEGACY_PROFILES=1
-          export MOZ_ALLOW_DOWNGRADE=1
-    '';
-
     home.packages = [ cfg.package ]
       ++ optional (any (p: p.withExternalGnupg) (attrValues cfg.profiles))
       pkgs.gpgme;
@@ -379,5 +365,11 @@ in {
           profile.extraConfig;
       };
     }));
+
+    # Mimic nixpkgs package environment for read-only profiles.ini management
+    home.sessionVariables = {
+      MOZ_LEGACY_PROFILES = 1;
+      MOZ_ALLOW_DOWNGRADE = 1;
+    };
   };
 }

From dd1602e306fec366280f5953c5e1b553e3d9672a Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Tue, 3 Sep 2024 17:10:10 -0400
Subject: [PATCH 2/2] firefox: set MOZ_* variables for legacy profiles.ini

This fixes app startup for installations that are not using
`wrapFirefox` (e.g. app from `homebrew`).

These variables are already set for nixpkgs firefox using wrapFirefox
wrapper.

This is most helpful for Darwin users. But it will help with an
(unusual) scenario where Home Manager is used in a non-NixOS environment
with the app managed by another package manager, or installed directly
from Mozilla builds.

Closes: #5717
---
 modules/programs/firefox/mkFirefoxModule.nix | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/programs/firefox/mkFirefoxModule.nix b/modules/programs/firefox/mkFirefoxModule.nix
index f22af019b066..69ef3c377e76 100644
--- a/modules/programs/firefox/mkFirefoxModule.nix
+++ b/modules/programs/firefox/mkFirefoxModule.nix
@@ -1012,6 +1012,12 @@ in {
           force = true;
         };
     }));
+
+    # Mimic nixpkgs package environment for read-only profiles.ini management
+    home.sessionVariables = {
+      MOZ_LEGACY_PROFILES = 1;
+      MOZ_ALLOW_DOWNGRADE = 1;
+    };
   } // setAttrByPath modulePath { finalPackage = wrapPackage cfg.package; });
 }
 
